### PR TITLE
TimeSeries HTML summary representations

### PR DESCRIPTION
**Pull request by mwhv2**
This pull request includes two summary representations in timeseriesbase: an HTML summary and an optional Plotly-based summary. The HTML summary can be accessed just like the Map summary (calling the TimeSeries object in a Jupyter notebook or by using the "quicklook" function). The plotly representation is accessed by calling the "summary" function. This is included to showcase the interactive functionality, but can be moved out of timeseriesbase and into a separate example in the future.

Two tests are included in the test_timeseriesbase for the HTML summary and quicklook functions. Also, three modified source files are included for goes, gbm, and lyra TimeSeries objects. The modifications include two possible new attributes ("_channel", and "_url") that provide the names and wavelengths (or energy ranges) for instrument-specific channels as well as a url to the instrument website. Both of these attributes are used in the summary representations and can be viewed using sample SunPy data for the above sources.

### CODE REVIEW

File: sunpy/timeseries/sources/fermi_gbm.py
Comment (nabobalis) [inline sunpy/timeseries/sources/fermi_gbm.py:25]: The removal of these line breaks, will need undoing.

File: sunpy/timeseries/timeseriesbase.py
Comment (nabobalis) [inline sunpy/timeseries/timeseriesbase.py:200]: There is no way to work this out outside of hardcoding it in the subclass?

Comment (mwhv2) [reply sunpy/timeseries/timeseriesbase.py:200]: Information about channel wavelengths/energies is typically not in the metadata or the fits headers. The exceptions seem to be Fermi and RHESSI where that information is the literal channel names. For other sources, this information is contained in the individual plot routines (either in axes labels or legend information). I may be missing something, but so far I have not found it in a somewhat common location.

**Review by dstansby — CHANGES_REQUESTED** [review]
I haven't looked at this in detail yet, but for now could you:

- Remove the `plotly` option? I think wheter or not we add this needs wider community discussion. It can always be added in a subsequent PR.
- Post a link to an example summary that's generated by this? It would be nice to take a look at it!

**Comment by mwhv2** [comment]
Here is a link to a Github Page with examples of the timeseries summary for sample Fermi data:
https://mwhv2.github.io/timeseries.html

**Comment by wafels** [comment]
See also #5834

**Comment by nabobalis** [comment]
We ended up merging a PR that created a repr_html for timeseries. You will need to rebase this on top and just wholesale replace the function.

How will this PR change when we merge in https://github.com/sunpy/sunpy/pull/5834? It replaces the underlying timeseries to be a astropy.timeseries.Timeseries. I would expect it not to matter?

**Comment by mwhv2** [comment]
Okay, I will rebase and update a few files in this PR. Everything should work with #5834.

File: sunpy/timeseries/timeseriesbase.py
Comment (nabobalis) [inline sunpy/timeseries/timeseriesbase.py:251]: What is the last `\` for?

File: sunpy/timeseries/timeseriesbase.py
Comment (nabobalis) [inline sunpy/timeseries/timeseriesbase.py:266]: Is there a problem if these are not in the main import block at the top?

**Comment by nabobalis** [comment]
I wonder what the best way to test this is.

File: sunpy/timeseries/timeseriesbase.py
Comment (mwhv2) [reply sunpy/timeseries/timeseriesbase.py:212]: When I have tested on the SunPy sample data, one or both are typically missing from eve level 0CS, rhessi, noaa, and norh subclasses.

File: sunpy/timeseries/timeseriesbase.py
Comment (mwhv2) [reply sunpy/timeseries/timeseriesbase.py:251]: Nothing in particular :). I will remove that in an update to this PR.

File: sunpy/timeseries/timeseriesbase.py
Comment (mwhv2) [reply sunpy/timeseries/timeseriesbase.py:266]: No, I will move them up to the top.

File: sunpy/timeseries/timeseriesbase.py
Comment (nabobalis) [reply sunpy/timeseries/timeseriesbase.py:214]: That is a problem but not for this PR.

**Comment by nabobalis** [comment]
For me, I think tests (not sure how best to do it for this) are the only thing missing from this PR.

Do you have an example image in a notebook for this?

**Comment by mwhv2** [comment]
Okay, I will brainstorm how to more effectively test these functions. Here are two example notebook images of goes and fermi sample data.
Goes_notebook
fermi_notebook

**Comment by nabobalis** [comment]
I wonder if we can just check the repr_html function runs for our all of our timeseries test data and assert maybe some of the text is correct?

**Comment by ayshih** [comment]
I'll point out that the analogous test for `GenericMap._repr_html_()` spot checks just a bit of the functionality due to the difficulty of testing HTML rendering:
https://github.com/sunpy/sunpy/blob/61bc3cf9bfe64ebe5eb9e884e2d95c5436d415c1/sunpy/map/tests/test_mapbase.py#L1149-L1177
You can copy the test for `GenericMap.quicklook()` in full.

**Comment by ConorMacBride** [comment]
If there was a stable way to produce an image from HTML, the image could be loaded into a matplotlib figure and passed to the figure tests.

**Comment by ayshih** [comment]
When I wrote #3951, we discussed adding chromium to the test suite to be able to test the rendered HTML, but we decided (at the time) that it was not worth the complexity.

**Comment by nabobalis** [comment]
There was a bug on main that broke the CI, you will need to rebase or merge in main again.

**Comment by mwhv2** [comment]
Okay, I have merged in main.

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Sorry this is taking so long for a second review @mwhv2

**Comment by nabobalis** [comment]
@mwhv2, would you be ok rebasing or merging in main again to get the tests running again?

**Comment by mwhv2** [comment]
Yeah, I have merged in sunpy main. No worries about the delay!

**Review by Cadair — APPROVED** [review]
Looks good to me.

It would be nice to swap to pytest-mock, but maybe better done in a follow up PR? What do you think @mwhv2?

File: sunpy/timeseries/tests/test_timeseriesbase.py
Comment (mwhv2) [reply sunpy/timeseries/tests/test_timeseriesbase.py:718]: This seems like an easy switch, but I am not familiar with the syntax of pytest-mock and am finding its implementation somewhat puzzling at the moment. It might be best that I update it in a future PR to avoid holding up this one.

**Comment by mwhv2** [comment]
The sample data example is now marked.

**Comment by nabobalis** [comment]
> The sample data example is now marked.

Thanks that has done it!

**Comment by nabobalis** [comment]
That oldest deps fail is real but confusing:


```
./../.tox/py38-oldestdeps/lib/python3.8/site-packages/sunpy/timeseries/tests/test_timeseriesbase.py::test_repr_html_ !!!!!!!!!!!!!!!!!!!! <ExceptionInfo TypeError("__init__() got an unexpected keyword argument 'filelocrepr'") tblen=3>

INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 327, in process_from_remote
INTERNALERROR>     rep = unserialize_report(eventname, kwargs["data"])
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 434, in unserialize_report
INTERNALERROR>     return runner.TestReport(**assembled_report(reportdict))
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 409, in assembled_report
INTERNALERROR>     reprentry = ReprEntry(
INTERNALERROR> TypeError: __init__() got an unexpected keyword argument 'filelocrepr'
[gw0] node down: <ExceptionInfo TypeError("__init__() got an unexpected keyword argument 'filelocrepr'") tblen=3>
[gw0] [ 87%] FAILED ../../.tox/py38-oldestdeps/lib/python3.8/site-packages/sunpy/timeseries/tests/test_timeseriesbase.py::test_repr_html_ 
Replacing crashed worker gw0

[gw2] linux Python 3.8.12 cwd: /home/runner/work/sunpy/sunpy/.tmp/py38-oldestdeps
WARNING: Failed to generate report: No data to report.

INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/_pytest/main.py", line 191, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/_pytest/main.py", line 247, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/dsession.py", line 115, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/dsession.py", line 138, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/dsession.py", line 258, in worker_runtest_protocol_complete
INTERNALERROR>     self.sched.mark_test_complete(node, item_index, duration)
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/scheduler/load.py", line 151, in mark_test_complete
INTERNALERROR>     self.node2pending[node].remove(item_index)
INTERNALERROR> KeyError: <WorkerController gw0>
!!!!!!!!!!!!!!!!!!!! <ExceptionInfo TypeError("__init__() got an unexpected keyword argument 'filelocrepr'") tblen=3>
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 327, in process_from_remote
INTERNALERROR>     rep = unserialize_report(eventname, kwargs["data"])
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 434, in unserialize_report
INTERNALERROR>     return runner.TestReport(**assembled_report(reportdict))
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 409, in assembled_report
INTERNALERROR>     reprentry = ReprEntry(
INTERNALERROR> TypeError: __init__() got an unexpected keyword argument 'filelocrepr'
!!!!!!!!!!!!!!!!!!!! <ExceptionInfo TypeError("__init__() got an unexpected keyword argument 'filelocrepr'") tblen=3>
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 327, in process_from_remote
INTERNALERROR>     rep = unserialize_report(eventname, kwargs["data"])
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 436, in unserialize_report
INTERNALERROR>     return runner.CollectReport(**assembled_report(reportdict))
INTERNALERROR>   File "/home/runner/work/sunpy/sunpy/.tox/py38-oldestdeps/lib/python3.8/site-packages/xdist/workermanage.py", line 409, in assembled_report
INTERNALERROR>     reprentry = ReprEntry(
INTERNALERROR> TypeError: __init__() got an unexpected keyword argument 'filelocrepr'
/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/multiprocessing/pool.py:265: ResourceWarning: unclosed running multiprocessing pool <multiprocessing.pool.ThreadPool state=RUN pool_size=2>
Exception ignored in: <function Pool.__del__ at 0x7f345a2f7e50>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/multiprocessing/pool.py", line 268, in __del__
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/multiprocessing/queues.py", line 362, in put
AttributeError: 'NoneType' object has no attribute 'dumps'
```

**Comment by nabobalis** [comment]
I think it's worth fixing the oldest deps test in a new PR to prevent holding this up.

**Comment by nabobalis** [comment]
Thanks for the PR and finishing this up through our many comments @mwhv2

**Pull request merged**
